ui/tests/integration/components/sync/secrets/page/overview-test.js

```diff
@@ -53,135 +53,263 @@ module('Integration | Component | sync | Page::Overview', function (hooks) {
+      assert.dom(overviewCard.title('Total destinations')).exists();
+      assert.dom(overviewCard.title('Total secrets')).exists();
+    });
+
+    module('Secrets by destination table', function () {
+      test('it should show the table with correct columns, data, badges, and actions', async function (assert) {
+        const { icon, name, badge, total, updated, actionToggle, action } = overview.table;
+        const updatedDate = dateFormat(
+          [new Date('2023-09-20T10:51:53.961861096-04:00'), 'MMMM do yyyy, h:mm:ss a'],
+          {}
+        );
+
+        await this.renderComponent();
+
+        assert
+          .dom(overviewCard.title('Secrets by destination'))
+          .hasText('Secrets by destination', 'Overview card title renders for table');
+        assert.dom(icon(0)).hasAttribute('data-test-icon', 'aws-color', 'Destination icon renders');
+        assert.dom(name(0)).hasText('destination-aws', 'Destination name renders');
+
+        assert.dom(badge(0)).hasText('All synced', 'All synced badge renders');
+        assert
+          .dom(badge(0))
+          .hasClass('hds-badge--color-success', 'Correct color renders for All synced badge');
+
+        assert
+          .dom(badge(1))
+          .doesNotExist('Status badge does not render for destination with no associations');
+
+        assert.dom(badge(2)).hasText('1 Unsynced', 'Unsynced badge renders');
+        assert.dom(badge(2)).hasClass('hds-badge--color-neutral', 'Correct color renders for unsynced badge');
+
+        assert.dom(total(0)).hasText('1', '# of external secrets renders');
+        assert.dom(updated(0)).hasText(updatedDate, 'Last updated datetime renders');
+
+        assert
+          .dom(total(1))
+          .hasText('0', '# of external secrets renders for destination with no associations');
+        assert
+          .dom(updated(1))
+          .hasText('—', 'Last updated placeholder renders for destination with no associations');
+
+        await click(actionToggle(0));
+        assert.dom(action('sync')).hasText('Sync secrets', 'Sync action renders');
+        assert.dom(action('details')).hasText('View synced secrets', 'View synced secrets action renders');
+      });
+
+      test('it should paginate the table', async function (assert) {
+        await this.renderComponent();
+
+        const { name, row } = overview.table;
+        assert.dom(row).exists({ count: 3 }, 'Correct number of table rows render based on page size');
+        assert.dom(name(0)).hasText('destination-aws', 'First destination renders on page 1');
+
+        await click(pagination.next);
+        await settled();
+        assert
+          .dom(overview.table.row)
+          .exists({ count: 3 }, 'New items are fetched and rendered on page change');
+        assert.dom(name(0)).hasText('destination-gcp', 'First destination renders on page 2');
+      });
+
+      test('it should show an empty state if there is an error fetching associations', async function (assert) {
```

Comment:
this test previously had a title that suggested an empty state would show when there were no destinations, but the content of the test made it seem like it was more about showing an empty state with a “check your policy” message, so i clarified that in the title of the test. 

that being said, i looked at the code and it looks like we show that “check your policy” message even when there isn’t a permissions error — we’d show it if ‘this.destinationMetrics’ is falsey. that seems misleading to me — maybe we should show a different empty state message instead?

Reply:
That seems like a good improvement to make! One thing to keep in mind is that the destinations call will fail with a specific message if the feature is not yet activated (which we can probably skip doing anything because the banner should already exist) and then other types of errors can show a catch-all error banner